dvc/dependency/__init__.py

```diff
@@ -48,11 +48,13 @@
 
 # NOTE: schema for dependencies is basically the same as for outputs, but
 # without output-specific entries like 'cache' (whether or not output is
-# cached, see -o and -O flags for `dvc run`) and 'metric' (whether or not
-# output is a metrics file and how to parse it, see `-M` flag for `dvc run`).
+# cached, see -o and -O flags for `dvc run`), 'metric' (whether or not output
+# is a metric file and how to parse it, see `-M` flag for `dvc run`) and
+# `store` (whether to push outputs to the remote)
 SCHEMA = output.SCHEMA.copy()
 del SCHEMA[BaseOutput.PARAM_CACHE]
 del SCHEMA[BaseOutput.PARAM_METRIC]
+del SCHEMA[BaseOutput.PARAM_STORE]
```

Comment:
These lines are not coupling outputs with dependencies. Maybe some refactoring is needed. But this is not related to this PR.